faas-samples/twitter-public/auto-respond-with-sentiment-analysis/README.md

```diff
@@ -0,0 +1,48 @@
+
+In this example, a reply will be posted on the consumer's tweet advising them to DM the brand. This DM will then flow into the Conversational Cloud where it can be handled by bots or agents.
+
+## Requirements
+
+- A LivePerson account with access to FaaS
+- A Twitter account
+- A [Twitter developer account](https://developer.twitter.com/en/apply-for-access)
+- A [Google developer service account](https://developers.google.com/identity/protocols/oauth2#serviceaccount) for the [Natural Language API](https://cloud.google.com/natural-language)
+
+## Setup
+
+1. Connect your Twitter account to LivePerson by following [this guide](https://docs.google.com/document/d/1CrMOi6exLPjXYSWVJvkvnwPNsUQaCUKsmP3OILAZTk8/edit?usp=sharing). If you don't have permission to access the document, please contact your LivePerson representative. Ensure that the public toggle for the Twitter account is flipped on.
+2. Set up a Filtered Stream rule in Houston so that only public tweets mentioning the brand will flow into the Conversational Cloud. As example rule for a brand "LPAirlines" can be seen below:
+
+```
+{
+	"autoStart": true,
+	"timeout": 20000,
+	"rules": [
+		{
+			"value": "@LPAirlines",
+			"tag": "brand mention"
+		}
+	]
+}
+```
+
+If you don't have access to Houston, or if Tweets matching the rule aren't displaying in the Conversational Cloud, please contact your LivePerson representative.
+
+3. [Create a Skill and Agent](https://developers.liveperson.com/tutorials-guides-getting-started-with-bot-building-deploy-the-bot.html) in LivePerson to handle Public Twitter messages.
+4. Create a Skill Routing rule in Houston for Public Twitter messages using [this guide](https://docs.google.com/document/d/1CrMOi6exLPjXYSWVJvkvnwPNsUQaCUKsmP3OILAZTk8/edit?usp=sharing). If you don't have permission to access the document, please contact your LivePerson representative.
+5. [Create a new FaaS function](https://developers.liveperson.com/liveperson-functions-getting-started.html) using the "Third Party Bots Custom Integration" event template and deploy the function. When creating the function, add 'api.twitter.com', 'language.googleapis.com, and 'oauth2.googleapis.com' to the Domain Whitelist.
+6. [Create a new third party bot](https://developers.liveperson.com/third-party-bots-custom-integration.html), selecting your FaaS function as the custom integration.
+7. Start the third party bot so its status is 'Online'.
+8. Replace the code in your FaaS function with the `autorespondwithsentimentanalysis.js` code in this repository.
+9. Add your Twitter Bearer Token (preceded by the word 'Bearer") to the FaaS secrets, using the key "TWITTER_TOKEN". The value will be in the format "Bearer {your_bearer_token}" (with no curly brackets).
+10. Add your Google Private Key (which will be in the downloaded JSON file available when creating a Google App Service Account) to the FaaS Secrets, using the key "GOOGLE_SECRET_KEY".
```

Comment:
Can we specify that the key is found under the property "private_key" here and is a string? I made the mistake of assuming that the whole JSON object should be included as the key and ran into some initial troubles with that. It's more obvious now that I'm re-reading this section, but I assume if I made the mistake, that others might as well.

Reply:
Nice! Happy that you were able to get it running without too much headache :) 

I'll update that step to make it clearer.

Reply:
@dahby ok the README has been updated.